### PR TITLE
Handle FunctionSpec.ReturnType properly for Plugin Framework

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/798

### CODE REVIEW

File: pf/Makefile [hunk 1]
```diff
@@ -58,4 +58,11 @@ tests/bin/pulumi-tfgen-testbridge::
...
 
-.PHONY: Test
+.PHONY: Test tidy
+
+tidy::
+	go mod tidy
+	cd tests && go mod tidy
+	cd tests/internal/randomshim && go mod tidy
+	cd tests/internal/tlsshim && go mod tidy
+	cd tests/testdatagen/genrandom && go mod tidy
```
Comment (jazzyfresh) [inline pf/Makefile:68]: too bad there isn't a recursive `go mod tidy`!

File: pf/internal/convert/func.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	switch {
+	case spec.Outputs != nil && spec.ReturnType == nil:
+		return spec.Outputs
+	case spec.ReturnType != nil && spec.Outputs == nil:
+		r := spec.ReturnType
+		switch {
+		case r.ObjectTypeSpec != nil && r.TypeSpec == nil:
+			return spec.ReturnType.ObjectTypeSpec
+		case r.TypeSpec != nil && r.ObjectTypeSpec == nil:
+			panic("TODO[pulumi/pulumi-terraform-bridge#796] TypeSpec is not supported yet by pf/tfbridge")
```
Comment (jazzyfresh) [inline pf/internal/convert/func.go:29]: This almost makes me miss [`java.lang.UnsupportedOperationException`](http://java.sun.com/javase/6/docs/api/java/lang/UnsupportedOperationException.html)

File: pf/internal/convert/func.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+
+func funcOutputs(spec *schema.FunctionSpec) *schema.ObjectTypeSpec {
```
[the remark this answers is not in the document]
Comment (jazzyfresh) [reply pf/internal/convert/func.go:19]: similarly, i would call this file `function.go`

File: pf/internal/convert/func.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		return spec.Outputs
+	case spec.ReturnType != nil && spec.Outputs == nil:
+		r := spec.ReturnType
+		switch {
+		case r.ObjectTypeSpec != nil && r.TypeSpec == nil:
+			return spec.ReturnType.ObjectTypeSpec
+		case r.TypeSpec != nil && r.ObjectTypeSpec == nil:
+			panic("TODO[pulumi/pulumi-terraform-bridge#796] TypeSpec is not supported yet by pf/tfbridge")
+		default:
+			panic("FunctionSpec.ReturnType should have either ObjectTypeSpec or TypeSpec")
```
Comment (jazzyfresh) [inline pf/internal/convert/func.go:31]: is there a case where they could have both? or is an exclusive-or situation

**Review by jazzyfresh — APPROVED** [review]

File: pf/internal/convert/func.go (hunk 4 above)
Comment (t0yv0) [reply pf/internal/convert/func.go:31]: The doc on the upstream type says it must be either-or.

**Pull request merged**